react-app/src/App.js

```diff
@@ -10,32 +12,39 @@ export function useUserLogged() {
 }
 
 function App() {
+    const history = useHistory()
 	const [userLoggedIn, setUserLoggedIn] = useState(null)
 
+	useEffect(() => {
+		authListener(signUserIn)
+	}, [])
+
+	function signUserIn(user) {
+		setUserLoggedIn(user)
+		if (user) {
+            history.replace('/')
+		}
+	}
+
 	function signUserOut() {
-		setUserLoggedIn(null)
+		signOut()
 	}
 
 	return (
-		<Router>
-			<UserLoginContext.Provider value={{userLoggedIn, signUserOut }}>
-				<Switch>
-					<Route exact path="/">
-						{userLoggedIn ? (
-							<DashboardPage />
-						): (
-							<Redirect to="/login" />
-						)}
-					</Route>
-					<Route path='/login'>
-						<LoginPage logUserIn={setUserLoggedIn}/>
-					</Route>
-					<Route path='/secure-login'>
-						<div>Password</div>
-					</Route>
-				</Switch>
-			</UserLoginContext.Provider>
-		</Router>
+		<UserLoginContext.Provider value={{userLoggedIn, signUserOut }}>
+			<Switch>
+				<Route exact path="/">
+					{userLoggedIn ? (
+						<DashboardPage />
+					): (
+						<Redirect to="/login" />
+					)}
```

Comment:
Let's discuss this one over the meeting - dynamic and declarative routing (especially redirects) like the ones in React as super great in day to day use, but they have some scalability issues (i.e. long term there are some down sides using this approach), so I want to address them specifically.

Reply:
- [ ] Discuss routing in React application